### DISCUSSION TITLE
What would be the best way to automate updating VueTorrent? I'm running Qbit in Docker, I use Watchtower, am on Ubuntu

### DISCUSSION THREAD

**Discussion by davin900** [VueTorrent/VueTorrent#1055] [Q&A]
Just curious... I've become pretty addicted to Watchtower handling my various Docker updates but VueTorrent I'm still updating manually every so often.

I see there are a few ways to bundle VueTorrent with Qbit via Docker Mods: https://github.com/arafatamim/linuxserver-io-mod-vuetorrent and https://github.com/gabe565/linuxserver-mod-vuetorrent

But it sounds like that would only update VueTorrent whenever the container restarts or updates. And Qbit itself doesn't update that often.

How would you go about accomplishing this? Thanks.

**Comment by Larsluph** [comment]
Personally, I created a service with a timer on my server to automatically pull changes from the `nightly-release` branch every hour.

`update-vuetorrent.service`
```ini
[Unit]
Description=Pull VueTorrent

[Service]
Type=simple
User=larsluph
Group=users
ExecStart=git -C /drives/HDD/appdata/qbit/VueTorrent pull
```

`update-vuetorrent.timer`
```ini
[Unit]
Description=Pull VueTorrent Timer

[Timer]
OnBootSec=1min
OnUnitActiveSec=1h

[Install]
WantedBy=timers.target
```

Note that I've used a clean qbit docker image (`linuxserver/qbittorrent`) and that I've manually pulled the Git repo using the command from the README (Installation > From Source)

**Reply by WDaan** [reply]
I guess either this, a cronjob, or relying on the hotio image?

**Comment by Larsluph** [comment]
[Dedicated wiki page](https://github.com/VueTorrent/VueTorrent/wiki/Keeping-VueTorrent-up%E2%80%90to%E2%80%90date-automatically) has been created for that purpose
